include/nlohmann/detail/output/serializer.hpp

```diff
@@ -398,6 +399,13 @@ class serializer
         std::size_t bytes_after_last_accept = 0;
         std::size_t undumped_chars = 0;
 
+        // copy string as-is if error handler is set to keep, and we don't want to ensure ASCII
+        if (error_handler == error_handler_t::keep && !ensure_ascii)
+        {
+            o->write_characters(s.data(), s.size());
```

Comment:
Just for me to understand, how would this behave exactly? If there is a `0×dc` byte for example, it will be escaped as `\334` octal string (or `\xdc` hex, or similar)?

I think the important thing is to not break the json format.

And also what about other UTF-8 accepted chars? Like `\b` or `\t` handled below: how will they be dumped in this case?

I have limited access these days (from mobile), and I don't know exactly the purpose of `ensure_ascii` and its default value. If you could provide some hint it would be helpful.

Thank you

Reply:
Oh, you're right! Just copying the input to the output is wrong here, because valid characters like LF that must be escaped to `\n` would not be escaped and the resulting JSON would be invalid. Thanks for noting. I will fix this.